pkg/backend/httpstate/stack.go

```diff
@@ -140,10 +144,60 @@ func newStack(ctx context.Context, apistack apitype.Stack, b *cloudBackend) (Sta
 		currentOperation: apistack.CurrentOperation,
 		tags:             apistack.Tags,
 		b:                b,
-		// We explicitly allocate the snapshot on first use, since it is expensive to compute.
+		hasRemoteConfig:  apistack.Config != nil,
 	}, nil
 }
-func (s *cloudStack) Ref() backend.StackReference                { return s.ref }
+func (s *cloudStack) Ref() backend.StackReference { return s.ref }
+
+// GetStackFilename returns the path to the stack file and a bool indicating if it's managed as a file.
+func (s *cloudStack) HasRemoteConfig() bool {
```

Comment:
Looking at how this is used I wonder if this should return a string of the remote environment name? We've got a load of places that call this then grab config.Imports[0] (with a fallback to <env>) but if this just returned the one environment name that we know is set then all that fallback code can be simplified.

Reply:
> the backend stack interface doesn't expose anything that ties this implementation to ESC

Sure but we then have a load of places that are tied to ESC for error and info messages. Either we should just embrace that this is tied to ESC in which case HasRemoteConfig can just return the environment we're tied to, or we should ensure that all the ESC handling is happening inside cloudStack.

Reply:
Hmm, let's sync on this next week to iron out the UX of the feedback etc.

Reply:
I've made the ESC environment available from the result of that backend stack method. I still think we will want remote config that's not tied to ESC, so I've kept the two from being synonymous.